src/main/java/baritone/behavior/ElytraBehavior.java

```diff
@@ -410,20 +439,69 @@ public void onTick(TickEvent event) {
 
         if (forceUseFirework || (!firework
                 && sinceFirework > 10
+                && Baritone.settings().elytraUseFireworks.value
                 && useOnDescend
                 && (ctx.player().posY < goingTo.y - 5 || start.distanceTo(new Vec3d(goingTo.x + 0.5, ctx.player().posY, goingTo.z + 0.5)) > 5) // UGH!!!!!!!
                 && currentSpeed < Baritone.settings().elytraFireworkSpeed.value)
         ) {
+            if (Baritone.settings().elytraInventory.value) {
+                final int firstFireworksInHotbar = firstFireworksInHotbar();
+                if (firstFireworksInHotbar == -1) {
+                    if (!swapToFireworksInInventory()) {
+                        logDirect("no fireworks");
+                        return;
+                    }
+                } else {
+                    ctx.player().inventory.currentItem = firstFireworksInHotbar;
+                }
+            }
```

Comment:
This looks *a lot* like `baritone.getInventoryBehavior().throwaway(true, ElytraBehavior::isBoostingFireworks)`

Reply:
^